src/Views/AppInfoView.vala

```diff
@@ -472,6 +488,43 @@ public class AppCenter.Views.AppInfoView : Adw.NavigationPage {
             }
         }
 
+        var supports = package_component.get_supports ();
+        for (int i = 0; i < supports.length; i++) {
+            var relation = supports[i];
+
+            string title, description, icon_name;
+
+            switch (relation.get_value_control_kind ()) {
+                // An input method for users to control software
+                case GAMEPAD:
+                    switch (relation.get_kind ()) {
```

Comment:
I personally don't like a switch inside a case, I think all the gamepad kind logic can be inside a method, then the code will be more organized and prepared to handle more kinds in the future.

Reply:
In this case "kind" just means "required", "recommended", or "supported" so I don't think there will be any more kinds in the future

Reply:
Ah sorry I misunderstood. Hows that? :)

Reply:
Cool, LGTM